tests/test_smoke.py

```diff
@@ -2763,6 +2763,36 @@ def generate_llm_test_command(prompt: str, expected_output: str) -> str:
     run_one_test(test)
 
 
+@pytest.mark.gcp
+def test_skyserve_spot_recovery():
```

Comment:
Shall we add some tests for non-recovery if the replica failed for some reason that is not preemption? For example, user bug?

Reply:
Good call, I'll work on this.

Reply:
Just added a nonrecovery test!